src/Compilers/VisualBasic/Portable/Lowering/ExpressionLambdaRewriter/ExpressionLambdaRewriter.vb

```diff
@@ -735,32 +761,40 @@ lSelect:
 
         ' Emit a call node
         Private Function [Call](receiver As BoundExpression, method As MethodSymbol, ParamArray params As BoundExpression()) As BoundExpression
-            Dim factoryArgs(0 To params.Length + 1) As BoundExpression
-            factoryArgs(0) = receiver
-            factoryArgs(1) = _factory.MethodInfo(method, _factory.WellKnownType(WellKnownType.System_Reflection_MethodInfo))
-            Array.Copy(params, 0, factoryArgs, 2, params.Length)
-            Return ConvertRuntimeHelperToExpressionTree("Call", factoryArgs)
+            Return _factory.Convert(
+                _expressionType,
+                    ConvertRuntimeHelperToExpressionTree(
+                        WellKnownMember.System_Linq_Expressions_Expression__Call,
+                        receiver,
+                        _factory.MethodInfo(method, _factory.WellKnownType(WellKnownType.System_Reflection_MethodInfo)),
+                        _factory.Array(_expressionType, params)))
         End Function
 
         ' Emit a Default node for a specific type
         Private Function [Default](type As TypeSymbol) As BoundExpression
-            Return ConvertRuntimeHelperToExpressionTree("Default", _factory.[Typeof](type, _factory.WellKnownType(WellKnownType.System_Type)))
+            Return _factory.Convert(
```

Comment:
Could you provide some context on why some calls to `ConvertRuntimeHelperToExpressionTree` are now wrapped in a `Convert` but not all?

Reply:
> Could you provide some context on why some calls to `ConvertRuntimeHelperToExpressionTree` are now wrapped in a `Convert` but not all?

Normally, ```Visit``` does this conversion. However, we are not getting here due to a ```Visit``` invocation and this conversion was applied during overload resolution, which we no longer perform. Therefore, the conversion is applied exp[licitly because types should match exactly.

Reply:
From offline discussion, `ConvertRuntimeHelperToExpressionTree` that are returned from a `Visit...` method are already converted, we only need to apply the conversion explicitly for helper/factory methods.
FWIW, I didn't observe any break in `CodeGenExprLambda.vb` (on either `net472` or `net9.0`) or in `Build.cmd -testCompilerOnly -test` when removing `Convert` in `Default` or `New`.

Reply:
I removed this conversion, apparently changes elsewhere made it unnecessary